src/api_gen.jl

```diff
@@ -1,27 +1,34 @@
 # Generate low-level api from METADATA
 
+using MsgPack
+import MsgPack: Extension, ExtensionType
+
 # Types defined by the api
-immutable NvimApiObject{N} <: NvimObject
+struct NvimApiObject{N} <: NvimObject
     client::NvimClient
     # TODO: use a fixarray or UInt64
     hnd::Vector{UInt8}
 end
 
 import Base.==
-=={N}(a::NvimApiObject{N},b::NvimApiObject{N}) = a.hnd == b.hnd
+==(a::NvimApiObject{N}, b::NvimApiObject{N}) where {N} = a.hnd == b.hnd
+
+NvimApiObject(c, e::Extension) = NvimApiObject{UInt8(e.type)}(c, e.data)
 
-NvimApiObject(c, e::Ext) = NvimApiObject{Int(e.typecode)}(c, e.data)
-#Not really module-interface clean, I know...
-MsgPack.pack{N}(s, o::NvimApiObject{N}) = MsgPack.pack(s, Ext(N, o.hnd))
+# Not really module-interface clean, I know...
+# TODO(smolck): Was just s with Any type, is this right?
+MsgPack.pack(s::IO, o::NvimApiObject{N}) where {N} = MsgPack.pack(s, Extension(N, o.hnd))
 
-symbolize(val::Dict) = Dict{Symbol,Any}([(symbolize(k),symbolize(v)) for (k,v) in val])
-symbolize(val::Vector{UInt8}) = symbol(bytestring(val))
-symbolize(val::ByteString) = symbol(val)
+symbolize(val::Dict) = Dict{Symbol,Any}([(symbolize(k), symbolize(v)) for (k, v) in val])
+# symbolize(val::Vector{UInt8}) = symbol(bytestring(val))
```

Comment:
Need to remove comment. And likewise for several others in this file.

Reply:
Fixed in https://github.com/bfredl/Neovim.jl/commit/97ae01b61b6696dcbf3ed8ccfd285658311cb256